plugins/woocommerce/src/Admin/Features/Settings.php

```diff
@@ -37,9 +37,37 @@ public function __construct() {
 		}
 
 		add_filter( 'woocommerce_admin_shared_settings', array( __CLASS__, 'add_component_settings' ) );
-		// Run this after the original WooCommerce settings have been added.
-		add_action( 'admin_menu', array( $this, 'register_pages' ), 60 );
-		add_action( 'init', array( $this, 'redirect_core_settings_pages' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_settings_editor_scripts' ) );
+	}
+
+	/**
+	 * Enqueue scripts for the settings editor.
+	 */
+	public function enqueue_settings_editor_scripts() {
+		$screen = get_current_screen();
+		if ( ! $screen || 'woocommerce_page_wc-settings' !== $screen->id ) {
+			return;
+		}
+
+		// Make sure the Settings Editor package is loaded.
+		wp_enqueue_script( 'wc-settings-editor' );
+
+		$script_handle          = 'wc-admin-edit-settings';
+		$script_path_name       = 'wp-admin-scripts/settings';
+		$script_assets_filename = WCAdminAssets::get_script_asset_filename( 'wp-admin-scripts', 'settings' );
+		$script_assets          = require WC_ADMIN_ABSPATH . WC_ADMIN_DIST_JS_FOLDER . 'wp-admin-scripts/' . $script_assets_filename;
+		$script_version         = WCAdminAssets::get_file_version( 'js', $script_assets['version'] );
+
+		wp_register_script(
```

Comment:
nit: Could we use `WCAdminAssets::register_script`? It will automatically take care of the versioning, dependencies, and translation.

https://github.com/woocommerce/woocommerce/blob/f0beaa988591feccbcac02b8a2e1a9300a7a4f18/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php#L460-L482

Reply:
I stuck this script in the same output of `wp-admin-scripts` which changes filepaths and names, so I wasn't able to use this function directly. I did this so that settings would not end up as `wc.settings` since its not a module for reuse.